model/llm/llm.go

```diff
@@ -24,9 +24,11 @@ import (
 
 // Model represents a LLM model accessed via a provider.
 type Model struct {
+	// id holds the full identifier, including the provider and attributes.
+	id string
```

Comment:
this could also be called "qualifier" to make it a bit less confusing

Reply:
or "selector"

Reply:
Would like to completely revamp how we address model IDs with the clean up of this hack. Will add it there.

Reply:
Added to #407